src/messaging/MessageCounterSync.cpp

```diff
@@ -38,17 +38,12 @@ namespace Messaging {
 
 CHIP_ERROR MessageCounterSyncMgr::Init(Messaging::ExchangeManager * exchangeMgr)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-
-    VerifyOrReturnError(exchangeMgr != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(exchangeMgr != nullptr, CHIP_ERROR_INCORRECT_STATE);
     mExchangeMgr = exchangeMgr;
 
     // Register to receive unsolicited Secure Channel Request messages from the exchange manager.
-    err = mExchangeMgr->RegisterUnsolicitedMessageHandlerForProtocol(Protocols::SecureChannel::Id, this);
-
-    ReturnErrorOnFailure(err);
-
-    return err;
+    // TODO: Register for specific message types, as CASE and PASE share the same protocol ID
+    return mExchangeMgr->RegisterUnsolicitedMessageHandlerForProtocol(Protocols::SecureChannel::Id, this);
```

Comment:
We should use RegisterUnsolicitedMessageHandlerForType here now to register only MCSP messages, otherwise, all PASE/CASE messages will be received by MessageCounterSync class

Reply:
The existing code is registering for the protocol type. I tried changing to only MCSP messages, but some tests are failing. That's why I added TODO so this can be tracked in an issue and fixed in a separate PR.

Reply:
I think that indicates a problem, by design, each protocol should only have one protocol delegate, since the message is moved not copied.

Now, there are two entities both register to receive SecureChannel message, it is really depending on registration order, the first one will be overridden by the second one.

Registering by different message type is also a workaround, but it should works, ideally, we should use SecureChannel protocol delegate to dispatch the messages, but we can leave it later as TODO

Reply:
Lets have an issue ticket to follow up here if the PASE messages are consumed by MCSP first, then PASE protocol should not receive the message.

Reply:
Yes, TODO bot will file the ticket once this PR is merged.